exercises/CI/introduction_to_ci/README.md

```diff
@@ -0,0 +1,154 @@
+
+> Continuous Integration is a software development practice where members of a team integrate their work frequently, usually each person integrates at least daily - leading to multiple integrations per day. Each integration is verified by an automated build (including test) to detect integration errors as quickly as possible. Many teams find that this approach leads to significantly reduced integration problems and allows a team to develop cohesive software more rapidly.
+- Martin Fowler: [https://martinfowler.com/articles/continuousIntegration.html](https://martinfowler.com/articles/continuousIntegration.html)
+
+Traditionally, large systems are very difficult to change and time consuming to verify. This results in new releases only being made a few times a year. These release are usually high risk, high anxiety, events due to their manual and error prone nature. CI is an automated process, this means, with careful management, that as the code base grows the time taken verify the code base should remain short (ideally just a few minutes). This has the knock on effect that new releases can be prepared and released frequently. Frequent, high quality, releases benefit the business and take the pressure off of the team.
+
+### Continuous Integration Servers
+Continuous integration servers are essentially glorified schedulers. They sit looking at your Version Control System (VCS) and when they see that a change has been introduced they do what ever they have been configured to do. E.g. to compile the new version of the code base and run a set of tests to see if it still works. Different CI servers may have different user interfaces, however they essentially all do the same thing and have a very common feature set. Therefore learn how to use one CI server and you should find that switching to another is a pretty easy process.
+
+## Exercise Learning Objectives
+- Introduction Concourse CI
+
+## Exercise
+**Note:** Before going any further do the following:
+- `cd YOUR_CLONE_OF_THIS REPO`
+- `source ./bin/env`
+- `cd `
+
+In this exercise you will be working using a CI server called Concourse. We've done the hard bit and have provided you with your very own Concourse installation. We've also provided you with a GIT server for your CI server to talk to. To start everything up, run: cic up 
+
+If all is well you should see the following:
+```
+  [OK] Environment cic'd up :)
+
+```
```

Comment:
Apparently even if all is not well, you'll see the same [OK] Environment cic'd up :) message

```
8055dd6791ed: Pull complete 
b749918e3aa2: Pull complete 
75c387007b66: Pull complete 
2b88a7cdad8f: Pull complete 
Digest: sha256:d6479f769d18b2d7d2f654a46d7715bd6a1ec5171870c7c8c58cdb1525a65815
Status: Downloaded newer image for lvlup/ci_course:0.5.3


Identity added: git-server/keys/id_rsa (git-server/keys/id_rsa)
Cloning into 'checkout'...
Warning: Permanently added 'git-server,172.18.0.3' (ECDSA) to the list of known hosts.
error: refs/heads/master does not point to a valid object!
fatal: git upload-pack: not our ref 5fc17da4781cb3df5f3c670b2007c10646a0f415
fatal: The remote end hung up unexpectedly
[ERROR] Failed to cic up the environment see above output for details

[OK] Environment cic'd up :)
```